assets/js/infinitescroll.js

```diff
@@ -1,115 +1,98 @@
-/* global maxPages */
-
-// Code snippet inspired by https://github.com/douglasrodrigues5/ghost-blog-infinite-scroll
-$(function ($) {
-    var currentPage = 1;
-    var pathname = window.location.pathname;
-    var $document = $(document);
-    var $result = $('.post-feed');
+/**
+ * Infinite Scroll
+ */
+
+(function(window, document) {
+    // next link element
+    var nextElement = document.querySelector('link[rel=next]');
+    if (!nextElement) return;
+
+    // post feed element
+    var feedElement = document.querySelector('.post-feed');
+    if (!feedElement) return;
+
     var buffer = 300;
 
     var ticking = false;
-    var isLoading = false;
+    var loading = false;
 
     var lastScrollY = window.scrollY;
     var lastWindowHeight = window.innerHeight;
-    var lastDocumentHeight = $document.height();
-
-    function onScroll() {
-        lastScrollY = window.scrollY;
-        requestTick();
-    }
-
-    function onResize() {
-        lastWindowHeight = window.innerHeight;
-        lastDocumentHeight = $document.height();
-        requestTick();
-    }
+    var lastDocumentHeight = document.documentElement.scrollHeight;
 
-    function requestTick() {
-        if (!ticking) {
-            requestAnimationFrame(infiniteScroll);
+    function onPageLoad() {
+        if (this.status === 404) {
+            window.removeEventListener('scroll', onScroll);
+            window.removeEventListener('resize', onResize);
+            return;
         }
-        ticking = true;
-    }
 
-    function sanitizePathname(path) {
-        var paginationRegex = /(?:page\/)(\d)(?:\/)$/i;
+        // append contents
+        var postElements = this.response.querySelectorAll('.post-card');
+        postElements.forEach(function (item) {
+            feedElement.appendChild(item);
+        });
 
-        // remove hash params from path
-        path = path.replace(/#(.*)$/g, '').replace('////g', '/');
+        // // push state
+        // window.history.pushState(null, document.title, nextElement.href)
```

Comment:
@zce were these comments left in for a reason? I think we probably want to avoid updating the history state in this implementation because it can create unexpected behaviour when sharing URLs

Reply:
There is no special reason. If you need to hold the state can be achieved in this way

Reply:
Can you remove them please? Sorry if my intention for the question wasn't clear 😄 We'd rather not have comments for unused functionality, especially when that functionality isn't directly usable by uncommenting.

Reply:
No problem